src/spikeinterface/curation/sortingview_curation.py

```diff
@@ -57,37 +57,55 @@ def apply_sortingview_curation(
     if "mergeGroups" in sortingview_curation_dict and not skip_merge:
         merge_groups = sortingview_curation_dict["mergeGroups"]
-        for mg in merge_groups:
+        for merge_group in merge_groups:
+            # Store labels of units that are about to be merged
+            labels_to_inherit = []
+            for unit in merge_group:
+                labels_to_inherit.extend(labels_dict.get(str(unit), []))
+            labels_to_inherit = list(set(labels_to_inherit))  # Remove duplicates
+
             if verbose:
-                print(f"Merging {mg}")
+                print(f"Merging {merge_group}")
             if unit_ids_dtype.kind in ("U", "S"):
                 # if unit dtype is str, set new id as "{unit1}-{unit2}"
-                new_unit_id = "-".join(mg)
+                new_unit_id = "-".join(merge_group)
+                curation_sorting.merge(merge_group, new_unit_id=new_unit_id)
             else:
                 # in this case, the CurationSorting takes care of finding a new unused int
-                new_unit_id = None
-            curation_sorting.merge(mg, new_unit_id=new_unit_id)
+                curation_sorting.merge(merge_group, new_unit_id=None)
+                new_unit_id = curation_sorting.max_used_id  # merged unit id
+                labels_dict[str(new_unit_id)] = labels_to_inherit
 
     # STEP 2: gather and apply sortingview curation labels
-
     # In sortingview, a unit is not required to have all labels.
     # For example, the first 3 units could be labeled as "accept".
     # In this case, the first 3 values of the property "accept" will be True, the rest False
-    labels_dict = sortingview_curation_dict["labelsByUnit"]
-    properties = {}
-    for _, labels in labels_dict.items():
-        for label in labels:
-            if label not in properties:
-                properties[label] = np.zeros(len(curation_sorting.current_sorting.unit_ids), dtype=bool)
+
+    # Initialize the properties dictionary
+    properties = {
+        label: np.zeros(len(curation_sorting.current_sorting.unit_ids), dtype=bool)
+        for labels in labels_dict.values()
+        for label in labels
+    }
+
+    # Populate the properties dictionary
     for u_i, unit_id in enumerate(curation_sorting.current_sorting.unit_ids):
-        labels_unit = []
-        for unit_label, labels in labels_dict.items():
-            if unit_label in str(unit_id):
-                labels_unit.extend(labels)
+        labels_unit = set()
+
+        # Check for exact match first
+        if str(unit_id) in labels_dict:
+            labels_unit.update(labels_dict[str(unit_id)])
+        # If no exact match, check if unit_label is a substring of unit_id (for string unit ID merged unit)
+        else:
+            for unit_label, labels in labels_dict.items():
+                if isinstance(unit_id, str) and unit_label in unit_id:
+                    labels_unit.update(labels)
```

Comment:
2. cast `unit_id` to str here:
```
unit_id_str = str(unit_id)
```
3. Use ar line 99:
```
    if unit_id_str in labels_dict: # which now will also contain string versions of the int new_unit_ids
```

Since we added all merged units to the `labels_dict`, we can get rid of the `else` (L102-105). 
In fact, by construction, the `prop_values` have `len(unit_ids)` size, so when setting the property at L110 we don't care if the unit ids are ints or strings!

@rkim48 do you agree that this could be an easier implementation?

Reply:
Yes thank you